File: 769_max-chunks-to-make-sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& v) {
        int w = 0, x = 0;
        for (int y = 0; y < v.size(); y++) {
            x += v[y] - y;
            w += (x == 0);
        }
        return w;
    }
};

File: 769_max-_chunks_to_make_sorted.java
class Solution {
    public int maxChunksToSorted(int[] a) {
        int b = 0, c = 0;
        for (int i = 0; i < a.length; i++) {
            c += a[i] - i;
            b += (c == 0) ? 1 : 0;
        }
        return b;
    }
}
